src/main.c

```diff
@@ -13,134 +13,332 @@
+                log_debug("Audio bit rate: %.1f kbps", (float)st->audio_bytes * 8 * 44100 / 2048 / st->audio_packets / 1000);
+                st->audio_packets = 0;
+                st->audio_bytes = 0;
+            }
+        }
+        break;
+    case NRSC5_EVENT_AUDIO:
+        if (evt->audio.program == st->program)
+            push_audio_buffer(st, evt->audio.data, evt->audio.count);
+        break;
+    case NRSC5_EVENT_SYNC:
+        st->audio_ready = 0;
+        break;
+    case NRSC5_EVENT_LOST_SYNC:
+        break;
+    case NRSC5_EVENT_ID3:
+        if (evt->id3.program == st->program)
+        {
+            if (evt->id3.title)
+                log_info("Title: %s", evt->id3.title);
+            if (evt->id3.artist)
+                log_info("Artist: %s", evt->id3.artist);
+            if (evt->id3.album)
+                log_info("Album: %s", evt->id3.album);
+            if (evt->id3.genre)
+                log_info("Genre: %s", evt->id3.genre);
+            if (evt->id3.ufid.owner)
+                log_info("Unique file identifier: %s %s", evt->id3.ufid.owner, evt->id3.ufid.id);
+            if (evt->id3.xhdr.param >= 0)
+                log_info("XHDR: %d %08X %d", evt->id3.xhdr.param, evt->id3.xhdr.mime, evt->id3.xhdr.lot);
+        }
+        break;
+    case NRSC5_EVENT_LOT:
+        if (st->aas_files_path)
+            dump_aas_file(st, evt);
+        log_info("LOT file: port=%04X lot=%d name=%s size=%d mime=%08X", evt->lot.port, evt->lot.lot, evt->lot.name, evt->lot.size, evt->lot.mime);
+        break;
+    }
 }
 
 static void help(const char *progname)
 {
-    fprintf(stderr, "Usage: %s [-v] [-q] [-l log-level] [-d device-index] [-g gain] [-p ppm-error] [-r samples-input] [-w samples-output] [-o audio-output -f adts|hdc|wav] [--dump-aas-files directory] frequency program\n", progname);
+    fprintf(stderr, "Usage: %s [-v] [-q] [-l log-level] [-d device-index] [-p ppm-error] [-g gain] [-r iq-input] [-w iq-output] [-o wav-output] [--dump-hdc hdc-output] [--dump-aas-files directory] frequency program\n", progname);
 }
 
-int main(int argc, char *argv[])
+static int parse_args(state_t *st, int argc, char *argv[])
 {
     static const struct option long_opts[] = {
         { "dump-aas-files", required_argument, NULL, 1 },
+        { "dump-hdc", required_argument, NULL, 2 },
         { 0 }
     };
-    int err, opt, gain = INT_MIN, ppm_error = 0;
-    unsigned int count, i, frequency = 0, program = 0, device_index = 0;
-    char *input_name = NULL, *output_name = NULL, *audio_name = NULL, *format_name = NULL, *files_path = NULL;
-    FILE *infp = NULL, *outfp = NULL;
-    input_t *input;
-    output_t *output;
-
-    while ((opt = getopt_long(argc, argv, "r:w:d:p:o:f:g:ql:v", long_opts, NULL)) != -1)
+    char *output_name = NULL, *audio_name = NULL, *hdc_name = NULL;
+    char *endptr;
+    int opt;
+
+    st->gain = -1;
+
+    while ((opt = getopt_long(argc, argv, "r:w:o:d:g:ql:v", long_opts, NULL)) != -1)
```

Comment:
```suggestion
    while ((opt = getopt_long(argc, argv, "r:w:o:d:p:g:ql:v", long_opts, NULL)) != -1)
```
This fixes the `-p` option.

Reply:
Oops. Thanks! Hopefully there aren't too many similar things from try to pry these PRs apart.